packages/slice-machine/src/features/settings/git/GitRepositoryConnectDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  Box,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogHeader,
+  FormInput,
+  Text,
+} from "@prismicio/editor-ui";
+import type { GitRepo, GitRepoSpecifier } from "@slicemachine/manager";
+import { Formik } from "formik";
+import type { FC, ReactNode } from "react";
+import { toast } from "react-toastify";
+
+import { gitProviderToConfig } from "@src/features/settings/git/GitProvider";
+import { useWriteAPIToken } from "@src/features/settings/git/useWriteAPIToken";
+
+type GitRepositoryConnectDialogProps = {
+  linkRepo: (repo: GitRepoSpecifier) => Promise<void>;
+  repo: GitRepo;
+  trigger: ReactNode;
+};
+
+export const GitRepositoryConnectDialog: FC<
+  GitRepositoryConnectDialogProps
+> = ({ linkRepo, repo, trigger }) => {
+  const providerConfig = gitProviderToConfig[repo.provider];
+  const { updateToken } = useWriteAPIToken({ git: repo });
+  return (
+    <Dialog size={{ width: 448, height: "auto" }} trigger={trigger}>
```

Comment:
The Dialog is closable while the token is being saved. Is it possible to disable closing it while it is loading? Or maybe we can disable the "X" and clicking outside the dialog completely, and only allow closing via the "Cancel" button?

Reply:
Is there a way to continue processing in the background after closing the modal? We'll need a toaster for confirmation or error messages in this scenario. 

Do we need to check the viability of the token?

Reply:
The second solution is good. 

In both solutions one and two, my only concern is the token's validity. Do we implement a way to check the token’s validity? 

Since it’s a temporary solution, I believe that’s not an issue anyway.

Reply:
If the second solution requires more time, I’m ok with releasing the quickest way of handling that. One or two.